### PR TITLE
fix 2918

### LINKED ISSUE #2918: Append with setValue and shouldDirty is not working correctly
**Describe the bug**
When appending values to field arrays with `setValue` and `{ shouldDirty: true }`, neither `dirtyFields` nor `isDirty` are being updated properly.

**To Reproduce**

1. Check `isDirty` and `dirtyFields` values
2. Click on the "add array field" or "add nested array field" button
3. Check that `isDirty` and `dirtyFields` values are not updated
4. Click on the "add array field" or "add nested array field" button again
5. Check that `isDirty` and `dirtyFields` values are updated but with previous values

**Codesandbox link (Required)**
https://codesandbox.io/s/nested-field-array-append-with-setvalue-and-shoulddirty-not-working-0m17e

**Expected behavior**
`isDirty` and `dirtyFields` should be updated right after triggering a `setValue` with `{ shouldDirty: true }`

### PR DESCRIPTION
**Pull request by bluebill1049**
## Fix
This PR is aimed to fix `dirtyFIelds` and `isDirty` state while using `setValue` to update FieldArray, fix #2918

## Codesandbox
CSB: https://codesandbox.io/s/nested-field-array-append-with-setvalue-and-shoulddirty-not-working-forked-367n5

### CODE REVIEW

**Comment by stephane439** [comment]
#

**Comment by stephane439** [comment]
Safir73@gmail.com

**Review by stephane439 — APPROVED** [review]

**Review by stephane439 — APPROVED** [review]

**Comment by stephane439** [comment]
@stephane439

File: src/logic/setFieldArrayDirtyFields.test.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import setFieldArrayDirtyFields from './setFieldArrayDirtyFields';
+
+describe('setFieldArrayDirtyFields', () => {
+  it('should set correctly dirtyFields', () => {
+    expect(
+      setFieldArrayDirtyFields(
```
[the remark this answers is not in the document]
Comment (stephane439) [reply src/logic/setFieldArrayDirtyFields.test.ts:6]: stephane439

[the remark this answers is not in the document]
Comment (stephane439) [reply src/logic/setFieldArrayDirtyFields.test.ts:6]:
````

**Review by stephane439 — APPROVED** [review]

**Review by beersmker — DISMISSED** [review]

**Review by JeromeDeLeon — APPROVED** [review]
LGTM! ❤️

**Comment by bluebill1049** [comment]
> LGTM! ❤️

Thanks @JeromeDeLeon for the review.

**Review by pmaier983 — APPROVED** [review]
LGTM 👍

**Comment by bluebill1049** [comment]
> LGTM 👍

Thanks @pmaier983's review.

**Pull request merged**